sendbuffer.go

```diff
@@ -106,28 +106,32 @@ func (buf *sendBuffer) sendLoop(w io.Writer) {
 
 func (buf *sendBuffer) send(b []byte, writeDeadline time.Time) (int, error) {
 	buf.muClosing.RLock()
-	defer buf.muClosing.RUnlock()
 	if buf.closing {
 		// Make it look like the write worked even though we're not going to send it
 		// anywhere (TODO, might be better way to handle this?)
+		buf.muClosing.RUnlock()
```

Comment:
To avoid having to manually unlock in all of these places, you could just create an inner function called `doSend` and handle the locking in the outer function.

Reply:
Oh nice idea. Made the change.